packages/react-testing/src/tests/root.test.tsx

```diff
@@ -0,0 +1,247 @@
+    it('removes the host DOM node in the DOM', () => {
+      const root = new Root(<div>Hello world</div>);
+      root.destroy();
+      expect(document.body.firstElementChild).toBeNull();
+    });
+  });
+
+  describe('#mount()', () => {
+    it('attaches a host div to the DOM', () => {
+      // eslint-disable-next-line no-new
+      new Root(<div>Hello world</div>);
+      expect(document.body.firstElementChild!).toBeInstanceOf(HTMLDivElement);
+    });
+
+    it('is called in the constructor to mount the element to a DOM node', () => {
+      // eslint-disable-next-line no-new
+      new Root(<div>Hello world</div>);
+      expect(document.body.firstElementChild!.innerHTML).toBe(
+        '<div>Hello world</div>',
+      );
+    });
+
+    it('throws an error when the component is already mounted', () => {
+      const root = new Root(<div>Hello world</div>);
+      expect(() => root.mount()).toThrowError(
+        /Attempted to mount a node that was already mounted/,
+      );
+    });
+
+    it('re-mountes an unmounted component', () => {
+      const root = new Root(<div>Hello world</div>);
+
+      root.unmount();
+      root.mount();
+      expect(document.body.firstElementChild!.innerHTML).toBe(
+        '<div>Hello world</div>',
+      );
+    });
+  });
+
+  describe('#setProps()', () => {
+    it('updates the props on the root node', () => {
+      function MyComponent(_props: {name: string}) {
+        return null;
+      }
+
+      const root = new Root(<MyComponent name="Gord" />);
+      expect(root.prop('name')).toBe('Gord');
+
+      root.setProps({name: 'goodforonefare'});
+      expect(root.prop('name')).toBe('goodforonefare');
+    });
+
+    it('does not unmount the component, but does trigger an update', () => {
+      const componentWillUnmount = jest.fn();
+      const componentDidUpdate = jest.fn();
+      class MyComponent extends React.Component<{name: string}> {
+        componentWillUnmount = componentWillUnmount;
+        componentDidUpdate = componentDidUpdate;
+
+        render() {
+          return null;
+        }
+      }
+
+      const root = new Root(<MyComponent name="Chris" />);
+      root.setProps({name: 'lemonmade'});
+
+      expect(componentWillUnmount).not.toHaveBeenCalled();
+      expect(componentDidUpdate).toHaveBeenCalled();
+    });
+  });
+
+  describe('#perform()', () => {
+    it('runs the callback in an act block', () => {
+      const root = new Root(<div />);
+
+      const queue = new Set<FirstArgument<Root<any>['perform']>>();
+      act.mockImplementation((perform: FirstArgument<Root<any>['perform']>) =>
+        queue.add(perform),
+      );
+
+      const perform = jest.fn();
+
+      root.perform(perform);
+      expect(perform).not.toHaveBeenCalled();
+
+      for (const queued of queue) {
+        queued();
+      }
+
+      expect(perform).toHaveBeenCalled();
+    });
+  });
+
+  describe('errors', () => {
+    const Thrower: React.ComponentType = () => {
```

Comment:
Should we test suspense style promise throwing? Or leave it till it's more mature

Reply:
We don't really use it since it doesn't support SSR so I think we should leave it out for now.